boskos/boskos.go

```diff
@@ -17,53 +17,79 @@ limitations under the License.
 package main
 
 import (
+	"bytes"
 	"encoding/json"
 	"flag"
 	"fmt"
+	"io"
 	"net/http"
 	"time"
 
 	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/boskos/common"
+	"k8s.io/test-infra/boskos/crds"
 	"k8s.io/test-infra/boskos/ranch"
+	"strings"
 )
 
-var configPath = flag.String("config", "resources.json", "Path to init resource file")
-var storage = flag.String("storage", "", "Path to persistent volume to load the state")
+var (
+	configPath  = flag.String("config", "config.yaml", "Path to init resource file")
+	storagePath = flag.String("storage", "", "Path to persistent volume to load the state")
+)
 
 func main() {
 	flag.Parse()
 	logrus.SetFormatter(&logrus.JSONFormatter{})
 
-	r, err := ranch.NewRanch(*configPath, *storage)
+	rc, err := crds.NewResourceClient()
+	if err != nil {
+		logrus.WithError(err).Fatal("unable to create a CRD client")
+	}
+
+	resourceStorage := crds.NewCRDStorage(rc)
+	storage, err := ranch.NewStorage(resourceStorage, *storagePath)
 	if err != nil {
-		logrus.WithError(err).Fatalf("Failed to create ranch! Config: %v, storage : %v", *configPath, *storage)
+		logrus.WithError(err).Fatal("failed to create storage")
 	}
 
-	http.Handle("/", handleDefault(r))
-	http.Handle("/acquire", handleAcquire(r))
-	http.Handle("/release", handleRelease(r))
-	http.Handle("/reset", handleReset(r))
-	http.Handle("/update", handleUpdate(r))
-	http.Handle("/metric", handleMetric(r))
+	r, err := ranch.NewRanch(*configPath, storage)
+	if err != nil {
+		logrus.WithError(err).Fatalf("failed to create ranch! Config: %v", *configPath)
+	}
+
+	boskos := http.Server{
+		Handler: NewBoskosHandler(r),
+		Addr:    ":8080",
```

Comment:
@krzyzacy we should probably make this configurable in the future

Reply:
Not a nit for this PR, just a note reading back through boskos, this PR is plenty large enough already :-)